// dep/src-gen/Dep.js
// Generated by N4JS transpiler; for copyright see original N4JS source file.

import 'n4js-runtime'

export class ClassInDep extends N4Object {
	constructor() {
		super();
	}
	methodOld() {
		console.log("#methodOld() in version 2.0.0");
	}
	methodNew() {
		console.log("#methodNew() in version 2.0.0");
	}
	static get n4type() {
		const $sym = Symbol.for('org.eclipse.n4js/reflectionInfo');
		if (this.hasOwnProperty($sym)) {
			return this[$sym];
		}
		const instanceProto = this.prototype, staticProto = this;
		return this[$sym] = new N4Class({
			name: 'ClassInDep',
			origin: 'ES6_Classes',
			fqn: 'tests/Dep/ClassInDep',
			n4superType: N4Object.n4type,
			allImplementedInterfaces: [],
			ownedMembers: [
				new N4Method({
					name: 'methodOld',
					isStatic: false,
					jsFunction: instanceProto['methodOld'],
					annotations: []
				}),
				new N4Method({
					name: 'methodNew',
					isStatic: false,
					jsFunction: instanceProto['methodNew'],
					annotations: []
				})
			],
			consumedMembers: [],
			annotations: []
		});
	}
}
